packages/yarnpkg-core/sources/httpUtils.ts

```diff
@@ -66,6 +85,10 @@ export async function request(target: string, body: Body, {configuration, header
   if (!configuration.get(`enableNetwork`))
     throw new Error(`Network access have been disabled by configuration (${method} ${target})`);
 
+  const networkConfig = getNetworkSettings(target, {configuration});
+  if (networkConfig?.config.get(`enableNetwork`) === false)
+    throw new Error(`Requests to '${target}' has been blocked by 'networkSettings["${networkConfig.glob}"].enableNetwork'`);
```

Comment:
I'd start with a less descriptive message without the glob pattern (just a generic `has been blocked because of your configuration settings`). It would make the `getNetworkSettings` interface simpler 🤔